src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,3222 @@
+            { cmmPrf = fromProto $ cdc ^. ProtoFields.prf
+            , cmmCredCounter = fromProto $ cdc ^. ProtoFields.credCounter
+            , cmmMaxAccounts = fromProto $ cdc ^. ProtoFields.maxAccounts
+            , cmmAttributes =
+                ( Map.fromAscList
+                    . fmap (\(tag, v) -> (AttributeTag $ fromIntegral tag, fromProto v))
+                    . Map.toAscList
+                )
+                    $ cdc ^. ProtoFields.attributes
+            , cmmIdCredSecSharingCoeff = fmap fromProto $ cdc ^. ProtoFields.idCredSecSharingCoeff
+            }
+instance FromProto Proto.AccountCredential where
+    type Output' Proto.AccountCredential = RawAccountCredential
+    fromProto ac =
+        case (ac ^? ProtoFields.normal, ac ^? ProtoFields.initial) of
+            (Just normal, Nothing) ->
+                NormalAC
+                    CredentialDeploymentValues
+                        { cdvPublicKeys = fromProto $ normal ^. ProtoFields.keys
+                        , cdvCredId = fromProto $ normal ^. ProtoFields.credId
+                        , cdvIpId = fromProto $ normal ^. ProtoFields.ipId
+                        , cdvPolicy = fromProto $ normal ^. ProtoFields.policy
+                        , cdvThreshold = fromProto $ normal ^. ProtoFields.arThreshold
+                        , cdvArData = fromProto $ normal ^. ProtoFields.arData
+                        }
+                    $ fromProto
+                    $ normal ^. ProtoFields.commitments
+            (Nothing, Just initial) ->
+                InitialAC
+                    InitialCredentialDeploymentValues
+                        { icdvAccount = fromProto $ initial ^. ProtoFields.keys
+                        , icdvRegId = fromProto $ initial ^. ProtoFields.credId
+                        , icdvIpId = fromProto $ initial ^. ProtoFields.ipId
+                        , icdvPolicy = fromProto $ initial ^. ProtoFields.policy
+                        }
+            _ -> error "oops"
+
+instance FromProto Proto.YearMonth where
+    type Output' Proto.YearMonth = YearMonth
+    fromProto yearMonth =
+        YearMonth
+            { ymYear = fromIntegral $ yearMonth ^. ProtoFields.year
+            , ymMonth = fromIntegral $ yearMonth ^. ProtoFields.month
+            }
+
+instance FromProto Proto.Policy where
+    type Output' Proto.Policy = Policy
+    fromProto p =
+        Policy
+            { pCreatedAt = fromProto $ p ^. ProtoFields.createdAt
+            , pValidTo = fromProto $ p ^. ProtoFields.validTo
+            , pItems = deMkAttributes $ p ^. ProtoFields.attributes
+            }
+      where
+        deMkAttributes =
+            Map.fromAscList
+                . fmap
+                    ( \(tag, value) ->
+                        ( AttributeTag $ fromIntegral tag
+                        , AttributeValue $ fromRight (error "oops") (S.runGet (S.getShortByteString 0) value) -- FIXME: Is n == 0 ok here? Do not use fromRight.
```

Comment:
Do not use fromRight.  I don't understand this parsing. The attribute value is a byte array. Why are you parsing 0 bytes?

Reply:
The documentation for the function specifies that this fails if fewer than `n` bytes are in the input? We have `n=0`in this case.

As I state, I know not to use `fromRight`, but utilizing `HasCallStack` has the added benefits of having the code compile before settling on an error handling strategy :-)